src/commands/topics.ts

```diff
@@ -166,6 +168,71 @@ async function editTopicConfig(topic: KafkaTopic): Promise<void> {
   editConfigForm.onDidDispose(() => disposable.dispose());
 }
 
+async function produceMessageFromFile(topic: KafkaTopic) {
+  const options: vscode.OpenDialogOptions = {
+    canSelectMany: false,
+    openLabel: "Select JSON file",
+    filters: {
+      "JSON files": ["json"],
+    },
+  };
+
+  const fileUri = await vscode.window.showOpenDialog(options);
+  if (fileUri && fileUri[0]) {
+    const filePath = fileUri[0].fsPath;
+    const message = JSON.parse(fs.readFileSync(filePath, "utf-8"));
+
+    if (!topic) {
```

Comment:
Please move this topic not truthy validation up to be the very first thing. No need to have done anything at all if this condition isn't met.

Reply:
nice catch! pushing this change up